sdk/core/Azure.Core/src/Pipeline/HttpPipelineMessage.cs

```diff
@@ -60,5 +60,76 @@ public void SetProperty(string name, object value)
 
             _properties[name] = value;
         }
+
+        public Stream? PreserveResponseContent()
```

Comment:
The name `PreserveResponseContent` makes me think this should be a bool.  I'm not sure of a better name though.  `AcquireResponseContentOwnership`?  `GetAndLockResponseContent`?  `ClaimResponseContent`?  Maybe @KrzysztofCwalina  has an idea for this one.

Reply:
ExtractResponseContent?